Admin page

RYin-28 opened the pull request:
![image](https://github.com/Jax-Donuts/Jax-website/assets/138041702/04c00ad4-d9ec-4760-9103-6bffbfa03ddc)

## Issue Name
Edit/Create product form

## Describe your changes
- Created a form based on the given specification.
- Added basic input validations to ensure specific fields aren't empty or invalid.
- Created a reusable "submit" style button.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] Do we need to implement analytics?

RYin-28 committed:
WIP
- Created new admin page (accessable at localhost:3000/admin)
- Created edit/create product form
- Form has some (basic) validations for most input fields

RYin-28 committed:
Admin Page pt 1 (Edit/Create Form)
- Created edit/create product form based on specifications
- Moved form to its own dedicated folder in admin folder to clean up main admin file
- Experimented in changing shade of red in header and home page header (color: "red" -> "red.8") for more consistancy
- Created a reusable "submit" style button (this may change depending on needs)
- Added submit button export to components index

drmicrochip97 commented on package-lock.json:
```diff
@@ -7,6 +7,7 @@
     "": {
       "name": "jax",
       "version": "0.1.0",
+      "hasInstallScript": true,
```
Hmmm, what is this? Can you research what the key is?

drmicrochip97 replied:
No need to delete.

RYin-28 replied:
It's a flag that indicates that the package has a preinstall, install, or post-install script.
https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json

drmicrochip97 commented on src/app/admin/edit-product-form/index.tsx:
```diff
@@ -0,0 +1,129 @@
+        <Paper radius="md" bg="#FFF5F5" p="xl">
+          <Box>
+            <Text align="center" fw={700}>
+              Edit/Create Product Form
```
Can remove "Form". The verbiage is mostly for us devs. The Title doesn't need it.

drmicrochip97 commented on src/components/button/submit-button.tsx:
```diff
@@ -0,0 +1,27 @@
+  },
+}))
+
+export function SubmitButton({ text }: { text: string }) {
```
move `{ text: string }` to an interface called Props. There should be other examples of how this is done in other components.

drmicrochip97 commented on src/components/button/submit-button.tsx:
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { Button, createStyles } from '@mantine/core'
+
+const useStyles = createStyles((theme) => ({
+  button: {
+    color: theme.black,
+    backgroundColor: theme.colors.red[6],
+    '&:hover': {
+      color: theme.black,
+      backgroundColor: theme.colors.red[8],
+    },
+  },
+}))
```
Move styles into `index.styles.ts` file

RYin-28 replied:
There was no `index.styles.ts` file unless you meant for me to make one. If that was the case, I made a `submit-button.styles.ts` file to follow the naming convention we already had in place.

drmicrochip97 commented on src/app/page.tsx:
```diff
@@ -45,7 +45,7 @@ export default function Home() {
                 style={{ justifyContent: 'center' }}
               />
-              <Title align="left" color="red" fw={700} size={80}>
+              <Title align="left" color="red.8" fw={700} size={80}>
```
interesting '.8'

RYin-28 replied:
Thought it would be better to just use Mantine's default color for better consistency.
https://mantine.dev/theming/colors/

drmicrochip97 commented on src/components/top-header/top-header.tsx:
```diff
@@ -13,7 +13,7 @@ export function TopHeader() {
             <Group spacing="xs">
               <Image src="/logo.png" alt="logo" width={32} height={32} />
-              <Title size="2rem" align="left" color="red" fw={700}>
+              <Title size="2rem" align="left" color="red.8" fw={700}>
```
For the Top Header component we should add the Admin button.

RYin-28 replied:
Added Admin button.

drmicrochip97 reviewed (commented):
Looks good, just some minor things.

RYin-28 committed:
PR #2 edits:
- Cleaned up admin page code.
- Created and moved submit button styles to its own file.
- Cleaned up code in submit button component.
- Added an Admin button to top header component.
- Made a small change to top header to optimize layout.

RYin-28 committed:
Last minute clean up.
